components/db/profile/profile.ts

```diff
@@ -68,13 +74,34 @@ export function useProfile() {
           dispatch({ updatingRep: false })
         }
       },
+      updateIsPrivate: async (isPrivate: string) => {
```

Comment:
We should keep using the public field for this setting as it's tied into our auth rules and used elsewhere on the site. A users profile is public if and only if `profile.public === true`. If it's false or unset, the profile is not public. It's ok to set `public: false` if it's currently unset.

The submitted value should be the value of the control prop passed to the modal, which we can update to a boolean. This way we aren't limited by the value types supported by buttons.

Reply:
uncoupling React Hook Form from this PR's Model seems to have cleared up a lot of my issues using booleans on this component, which in turn has let me rewire this back to profile.public, etc